core/packages/PackageInfo.h

```diff
@@ -182,10 +182,11 @@ class PackageInfo {
 
     StrictDependenciesLevel strictDependenciesLevel = StrictDependenciesLevel::None;
 
-    // Map from pkg -> {list of files in this package that reference pkg, whether this package is missing an import
-    // for pkg and whether importing this package would cause a modularity error}
-    UnorderedMap<core::packages::MangledName, std::pair<UnorderedSet<core::FileRef>, PackageReferenceInfo>>
-        referencedPackages = {};
+    // Map from file -> [{a package referenced by file, whether the import is missing, and whether importing it would be
+    // a modularity error}]
+    // TODO(neil): once we track a list of files in this package, we can `.reserve(files.size())` in the constructor
+    UnorderedMap<core::FileRef, std::vector<std::pair<core::packages::MangledName, PackageReferenceInfo>>>
+        packagesReferencedByFile;
```

Comment:
```suggestion
    UnorderedMap<core::FileRef, std::vector<std::pair<MangledName, PackageReferenceInfo>>> packagesReferencedByFile;
```

likewise, shortening this name allows a single-line definition.